arwen/host/execution.go

```diff
@@ -411,6 +421,21 @@ func (host *vmHost) callSCMethodIndirect() error {
 	return nil
 }
 
+func (host *vmHost) callBuiltinFunction(input *vmcommon.ContractCallInput) error {
+	_, _, metering, output, runtime, _ := host.GetContexts()
+
+	valueToSend, gasConsumed, err := host.blockChainHook.ProcessBuiltInFunction(input)
+	if err != nil {
+		metering.UseGas(gasConsumed)
+		return err
+	}
+
+	scAccount, _ := output.GetOutputAccount(runtime.GetSCAddress())
```

Comment:
seeing this I think we have to make sure who gets the money, now it is correct, as only in claimdeveloperrewards - and the sender can get it. however processbuiltinfunction might have to return a vmoutput ? not know, but we should discuss it

Reply:
If a builtin function transfers money that aren't necessarily destined to the caller of the function, then yes, more info must be returned to the VM. The cleanest alternative I can think of right now is to return a map of `OutputAccount` objects, which can be merged by the VM on the `outputContext`. This might be better than a full `VMOutput`, although `ReturnData` would also be very useful. Yes, this needs more discussion.